src/realm/query_expression.hpp

```diff
@@ -4201,8 +4215,25 @@ class Compare : public Expression {
                                                  m_left->description(state));
         }
         else {
-            return util::serializer::print_value(m_left->description(state) + " " + TCond::description() + " " +
-                                                 m_right->description(state));
+            std::string value = m_right->description(state);
+            if constexpr (std::is_same_v<TCond, Equal>) {
```

Comment:
I don't know if this is the right place to do this? I would have expected `Value::value_to_string()` to handle this. Either way, we still need to handle a TypedLink from Mixed, since `get_target_table()` returns nullptr for that case.

Reply:
I think I have a better solution now.